src/java.base/share/classes/java/security/Security.java

```diff
@@ -47,6 +47,10 @@
  * implementation-specific location, which is typically the properties file
  * {@code conf/security/java.security} in the Java installation directory.
  *
+ * @implNote If the master security properties file is missing, the JDK
+ * implementation will throw an {@code InternalError} when initializing the
+ * {@code Security} class.
```

Comment:
Is "master security properties file" the right term here? The only use of "master" in this class seems to be a deprecated method.

I'm also not sure about the word "missing" as there can be many reasons why opening a file may fail, e.g. something changes the permissions or file owner so that it can't be opened for example.

The implNote need not be specific on the exception that is thrown, it could say "will throw an unspecified exception ..."

Reply:
The opening line of the java.security file denotes it as the "master security properties file". I think it works well but open to suggestions. Maybe we can re-use the terminology from previous paragraph.

Fair point about the "missing" and error clause -- perhaps:
 ```
@implNote If the file containing the default values of security properties fails to load,
 * the JDK implementation will throw an unspecified error when initializing the
 * {@code Security} class.
```

Reply:
I think if you just said "If this properties file ..." it would be sufficient, as the previous paragraph provided enough context as to what you are referring to.

Reply:
Btw the current code does not handlmissing or unreadable overwrite files very well. PropsLoaded will be true even when the replacing overwrite file could not be loaded (resulting in an empty list).

Reply:
it's a fair point @ecki -- the 2nd `loadedProps = true;` call is redundant since the code block is not executed unless the `security.overridePropertiesFile` is set to true by reading the props from default/master file.

one option here might be to delay the re-initialization of `props` until we successfully load the override security props file

i.e. push this code block down 
```
            if (overrideAll) {
                props = new Properties();
                if (sdebug != null) {
                    sdebug.println
                        ("overriding other security properties files!");
                }
            }
```

I'll log another JBS issue to track that.

Reply:
@ecki  -- Logged JDK-8292297 to track the issue you raised